### PR TITLE
[feat] Article page

### PR DESCRIPTION
**Pull request by Tristancalard**

### CODE REVIEW

File: frontend/config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const API_URL = process.env.NEXT_PUBLIC_DEBUG === 'true' ? 'http://localhost:3000' : 'apiUrl_hub';
+
+export{
+    API_URL,
+};
```
Comment (alanschnegg) [inline frontend/config.js]: 'apiUrl_hub' should be in .env

File: frontend/pages/article.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    const options = {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body:
+            JSON.stringify({
+                name: name,
+                content: content,
+            }),
```
Comment (alanschnegg) [inline frontend/pages/article.tsx:47]: STOP STRINGIFY YOUR BODY 
Body should be a json object

File: frontend/pages/article.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    const options = {
+        method: 'PATCH',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body:
+            JSON.stringify({
+                name: name,
+                content: content,
+            }),
```
Comment (alanschnegg) [inline frontend/pages/article.tsx:62]: Same

File: frontend/pages/article.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        setValue(newValue);
+    };
+
+    const [value2, setValue2] = useState('');
+    useEffect(() => {
+        setValue2(localStorage.getItem(STORAGE_KEY2) || '');
+    }, []);
+    const update2 = (newValue: string) => {
+        localStorage.setItem(STORAGE_KEY2, newValue);
+        setValue2(newValue);
```
Comment (alanschnegg) [inline frontend/pages/article.tsx]: Better variable name, need explicit one

File: frontend/pages/article.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                        onClick={async () => {
+                            const response = await addArticle(`Article_${new Date().toISOString()}`, value);
+                            const newarticle = await response.json();
+                            setArticlesData(old => [...old, newarticle]);
+                            setOpen(false);
+                            update('');
+                        }}
+                        className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded mt-6" >Add Article</button>
+                </Box>
+            </Modal>
```
Comment (alanschnegg) [inline frontend/pages/article.tsx]: Move this to a component

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: backend/src/app.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ import authRouter from './routes/auth/auth.js';
 import inventoryRouter from './routes/inventory/inventory.js';
 import projectRouter from './routes/projects/projects.js';
 import usersRouter from './routes/users/router.users.js';
+import articleRouter from './routes/article/router_article.js';
```
Comment (RainproDash) [inline backend/src/app.ts]: file should be name with . not with _

**Review by RainproDash — CHANGES_REQUESTED** [review]

**Review by RainproDash — APPROVED** [review]

File: backend/src/app.ts [hunk 7]
```diff
@@ -12,6 +12,7 @@ import authRouter from './routes/auth/auth.js';
 import inventoryRouter from './routes/inventory/inventory.js';
 import projectRouter from './routes/projects/projects.js';
 import usersRouter from './routes/users/router.users.js';
+import articleRouter from './routes/article/router.article.js';
```
Comment (thmarinho) [inline backend/src/app.ts]:
```suggestion
import articleRouter from './routes/article/router.articles.js';
```

File: backend/src/config/db.ts [hunk 8]
```diff
@@ -5,6 +5,7 @@ import { Project } from '../entity/projects.js';
 import { User } from '../entity/user.js';
 import { DataSource, DataSourceOptions } from 'typeorm';
 import { SeederOptions } from 'typeorm-extension';
+import { Article } from '../entity/article.js';
```
Comment (thmarinho) [inline backend/src/config/db.ts]: Move it upward w/ other entities

File: frontend/components/dialog.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
+import React, { Fragment } from 'react';
+import dynamic from 'next/dynamic';
+import 'react-quill/dist/quill.snow.css';
+import { Dialog, Transition} from '@headlessui/react';
```
Comment (thmarinho) [inline frontend/components/dialog.tsx]:
```suggestion
import React, { Fragment } from 'react';
import dynamic from 'next/dynamic';
import { Dialog, Transition} from '@headlessui/react';
import 'react-quill/dist/quill.snow.css';

```

File: frontend/pages/article.tsx [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    name: string;
+    content: string;
+}
+
+function deleteArticle(id: number) {
+    const options = {
+        method: 'DELETE',
+        headers: {
+            'Content-Type': 'application/json',
+        },
```
Comment (thmarinho) [inline frontend/pages/article.tsx:32]:
```suggestion
```

File: frontend/pages/article.tsx [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+    };
+    fetch(`${API_URL}/article/${id}`, options);
+}
+
+async function addArticle(name: string, content: string){
+    const options = {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
```
Comment (thmarinho) [inline frontend/pages/article.tsx:42]:
```suggestion
```

File: frontend/pages/article.tsx [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+        body:
+            JSON.stringify({
+                name: name,
+                content: content,
+            }),
```
Comment (thmarinho) [inline frontend/pages/article.tsx:47]:
```suggestion
{
  name,
  content,
}
```

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by alanschnegg — APPROVED** [review]

**Review by thmarinho — APPROVED** [review]

**Pull request merged**